tensorflow/contrib/receptive_field/python/util/receptive_field.py

```diff
@@ -304,13 +308,72 @@ def _get_effective_padding_node_input(stride, padding,
   return stride * effective_padding_output + padding
 
 
-def compute_receptive_field_from_graph_def(graph_def, input_node, output_node):
+class ReceptiveField:
+  """
+  Receptive field of a convolutioanl neural network.
+
+  Args:
+    size: Receptive field size.
+    stride: Effective stride.
+    padding: Effective padding.
+  """
+  def __init__(self, size, stride, padding):
+    self.size = np.asarray(size)
+    self.stride = np.asarray(stride)
+    self.padding = np.asarray(padding)
+
+  def compute_input_coordinates(self, x, axis=None):
+    """
+    Computes the center of the receptive field that generated a feature.
+
+    Args:
+      x: Coordinate of the feature.
+      axis: The dimensions for which to compute the input coordinates.
+
+    Returns:
+      y: Center of the receptive field that generated the feature.
+    """
+    # Use all dimensions
+    if axis is None:
+      axis = range(self.size.size)
+    # Ensure axis is a list because tuples have different indexing behavior
+    axis = list(axis)
+    return - self.padding[axis] + x * self.stride[axis] + \
+      (self.size[axis] - 1) / 2
+
+  def compute_feature_coordinates(self, y, axis=None):
+    """
+    Computes the position of a feature given the center of a receptive field.
+
+    Args:
+      y: Center of the receptive field.
+      axis: The dimensions for which to compute the feature coordinates.
+
+    Returns:
+      x: Coordinate of the feature.
+    """
+    # Use all dimensions
+    if axis is None:
+      axis = range(self.size.size)
+    # Ensure axis is a list because tuples have different indexing behavior
+    axis = list(axis)
+    return (y + self.padding[axis] + (1 - self.size[axis]) / 2) / \
+      self.stride[axis]
+
+  def __iter__(self):
+    return iter(np.concatenate([self.size, self.stride, self.padding]))
+
+
+def compute_receptive_field_from_graph_def(graph_def, input_node, output_node,
+                                           stop_propagation=None):
   """Computes receptive field (RF) parameters from a GraphDef object.
 
   Args:
     graph_def: GraphDef object.
     input_node: Name of the input node from graph.
     output_node: Name of the output node from graph.
+    stop_propagation: List of operation  or scope names for which to stop the
```

Comment:
I understand the logic for the need of stop_propagation. I guess it is useful in order to compute receptive field information for specific paths of a network. (in other words, in a CNN there are multiple ways to go from input to output; stop_propagation supports measuring RF information for only part of the network).
Are there some other cases where it is useful? Anyway, if you could just one or a couple more sentence to explain to the not-so-familiar reader when stop_propagation can be useful, it would be perfect.